### PR TITLE
Improvements for Quest Log window

### PR DESCRIPTION
**Pull request by ArseniyShestakov**
So here is result of few days of my work:
* New background graphics with source in "[vcmi-assets](https://github.com/vcmi/vcmi-assets/tree/develop/Interface/Quest%20Log%20window)" repository.
* Everything displayed correctly, minimap change level automatically.
* Components of quest displayed under description.
* Quest Log button in right panel now disabled on maps that don't have quests.

Though there is still things need to be done for quest log:
* We'll need to decide if we want to see completed quest in log and how to divide completed and active, etc.
* Quest button have to be disabled if player don't have (active?) quests.
* There need some work on CTextBox itself as with slider it's looks bad.
* Would be also useful to rewrite CComponentBox to have more control of how it's looks like.
* Some code written before me also need improvements.

@alexvins also suggested feature that would give a hint if player have some components already within hero/town/etc and I'll likely implement it.

Will really appreciate ideas/suggestions on code, design or features.. 

Some screenshots: [- ONE -](https://cloud.githubusercontent.com/assets/9513350/6198083/77fffdd8-b403-11e4-9a41-caf6b9d4a06d.png) and [- TWO - ](https://cloud.githubusercontent.com/assets/9513350/6198084/7800e4f0-b403-11e4-8bb2-3e59d484eb26.png)
Related issues: [#1988](http://bugs.vcmi.eu/view.php?id=1988) [#2022](http://bugs.vcmi.eu/view.php?id=2022)

### CODE REVIEW

File: client/windows/CQuestLog.cpp [hunk 1]
```diff
@@ -120,6 +124,7 @@ CQuestLog::CQuestLog (const std::vector<QuestInfo> & Quests) :
 	CWindowObject(PLAYER_COLORED | BORDERED, "questDialog.pcx"),
```
Comment (alexvins) [inline client/windows/CQuestLog.cpp]: Resource names should be w/o extension or at least with right one - "png"

**Comment by DjWarmonger** [comment]
One thing: instead of "bring something to Seer's Hut" it should read "Bring something to Xavier The Seer". This could be somewhat easier to remember / tell apart.

**Comment by ArseniyShestakov** [comment]
@DjWarmonger done. :-)

**Comment by DjWarmonger** [comment]
Okay, so you want to merge that branch or continue work?

**Comment by ArseniyShestakov** [comment]
I'm still want to find some solution for

**Comment by ArseniyShestakov** [comment]
From one point of view it's may be worth to hide completed quests, but then it's would be better to show them as this is still useful information. May be stike quest label or somehow make it look different?

Also do we really need quest log buttons on Hero Exchange window?

**Comment by ArseniyShestakov** [comment]
Ooops. Just find out there use case where they may be useful as exchange take movement points so player might want to know what quests items both heroes have.

**Comment by DjWarmonger** [comment]
How about additional switch to show/hide complete quest?

**Comment by ArseniyShestakov** [comment]
Any idea how it's can looks like from visual point of view?
May be checkbox icon with like in "System Options" with text of "Show completed quests"?

I can do little of graphics editing, but when it's come to new stuff it's hard.

**Comment by DjWarmonger** [comment]
Dismiss buitton from creature / hero window?

**Comment by alexvins** [comment]
Vote for checkbox.

**Comment by ArseniyShestakov** [comment]
Done it with checkbox. At moment it's looks like that:
questlog_hide

If everyone think code in good shape it's can be merged now. I'll likely back to Quest Log window later, but first I need to improve adventure map objects and device what to do with keymasters.

**Pull request merged**